### PR TITLE
BottleLoader: check version mismatch

### PR DESCRIPTION
**Pull request by xu-cheng**

### CODE REVIEW

File: Library/Homebrew/cmd/cleanup.rb [hunk 1]
```diff
@@ -83,7 +83,9 @@ def cleanup_cache
         next
       end
 
-      if f.version > version || ARGV.switch?('s') && !f.installed? || bottle_file_outdated?(f, file)
+      file_is_stale = PkgVersion === version ? f.pkg_version > version : f.version > version
+
+      if file_is_stale || ARGV.switch?('s') && !f.installed? || bottle_file_outdated?(f, file)
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/cleanup.rb]: I realise the existing logic is also hairy but might be worth splitting this out into e.g. a few `return`s so it's easier to follow.

File: Library/Homebrew/cmd/cleanup.rb [hunk 2]
```diff
@@ -83,7 +83,9 @@ def cleanup_cache
         next
       end
 
-      if f.version > version || ARGV.switch?('s') && !f.installed? || bottle_file_outdated?(f, file)
+      file_is_stale = PkgVersion === version ? f.pkg_version > version : f.version > version
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/cleanup.rb]: Personally I find the ternary operator hard to follow in cases like this.

**Comment by mikemcquaid** [comment]
A few comments but :+1: overall!

File: Library/Homebrew/cmd/cleanup.rb (hunk 2 above)
Comment (xu-cheng) [inline Library/Homebrew/cmd/cleanup.rb]: Updated to use `if .. else ..`

File: Library/Homebrew/cmd/cleanup.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/cmd/cleanup.rb]: I didn't change anything so far as I realize this whole method need to do a serious change. (e.g. better matching for name, handling resources etc.) So I intend to address this in future PR.

**Comment by mikemcquaid** [comment]
Cool. :+1: once green.
